File: core/MergeTwoSortedLists/MergeTwoSortedLists.java
package com.dbc;

public class MergeTwoSortedLists {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode prev = new ListNode();
        ListNode res = prev;
        while (list1 != null && list2 != null) {
            ListNode node = null;
            if (list1.val <= list2.val) {
                node = list1;
                list1 = list1.next;
            } else {
                node = list2;
                list2 = list2.next;
            }
            prev.next = node;
            prev = prev.next;
        }
        prev.next = list1 != null ? list1 : list2;

        return res.next;
    }
}

File: core/MergeTwoSortedLists/MergeTwoSortedLists.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def mergeTwoLists(list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
    prev = ListNode()
    res = prev
    while list1 and list2:
        if list1.val <= list2.val:
            node = list1
            list1 = list1.next
        else:
            node = list2
            list2 = list2.next
        prev.next = node
        prev = prev.next

    node = list1 if list1 else list2
    prev.next = node
    return res.next


